### PR TITLE
src: copier_hifi: update apply_attenuation function

### PR DESCRIPTION
**Pull request by dnikodem**
The function applying the attenuation should be based on write pointer and the current number of copied data in the current systick.

### CODE REVIEW

File: src/audio/copier/copier_hifi.c [hunk 1]
```diff
@@ -31,7 +31,8 @@ int apply_attenuation(struct comp_dev *dev, struct copier_data *cd,
 	ae_valign uu = AE_ZALIGN64();
 	ae_valign su = AE_ZALIGN64();
 	int remaining_samples = frame * audio_stream_get_channels(&sink->stream);
-	uint32_t *dst = audio_stream_get_rptr(&sink->stream);
+	uint32_t bytes = frame * audio_stream_frame_bytes(&sink->stream);
```
Comment (btian1) [inline src/audio/copier/copier_hifi.c:34]: Please change copier_generic code as well.

**Review by btian1 — COMMENTED** [review]
I would suggest add a test case in CI or QB to check attenuation, recently, this is the second patch for attenuation, first is: [#7761](https://github.com/thesofproject/sof/pull/7761)

We don't know how to verify this fix,, it just looks reasonable.

File: src/audio/copier/copier_hifi.c (hunk 1 above)
Comment (dnikodem) [reply src/audio/copier/copier_hifi.c:34]: Good point, I will update this. Thanks

**Comment by dnikodem** [comment]
SOFCI TEST

File: src/include/sof/audio/audio_stream.h [hunk 2]
```diff
@@ -773,6 +773,29 @@ audio_stream_rewind_bytes_without_wrap(const struct audio_stream __sparse_cache
...
+ */
+static inline uint32_t
+*audio_stream_rewind_wptr_by_bytes(const struct audio_stream __sparse_cache *source,
+				       const uint32_t bytes)
+{
+	void *wptr = audio_stream_get_wptr(source);
+	int to_begin = audio_stream_rewind_bytes_without_wrap(source, wptr);
+
+	assert((intptr_t)wptr >= (intptr_t)source->addr);
+	assert((intptr_t)source->end_addr >= (intptr_t)wptr);
```
Comment (lyakh) [inline src/include/sof/audio/audio_stream.h:791]: actually should be strictly `>`?

**Review by lyakh — APPROVED** [review]

File: src/include/sof/audio/audio_stream.h (hunk 2 above)
Comment (dnikodem) [reply src/include/sof/audio/audio_stream.h]: Fixed, thanks

**Comment by dnikodem** [comment]
> I would suggest add a test case in CI or QB to check attenuation, recently, this is the second patch for attenuation, first is: [#7761](https://github.com/thesofproject/sof/pull/7761)
> 
> We don't know how to verify this fix,, it just looks reasonable, do you know whether QB test cover attenuation or ready for add?

I talked with our validation team - they will try to find a way to validate attenuation in our CI/QB (or they will confirm that it is validate properly). Thanks

**Review by tmleman — APPROVED** [review]

**Review by btian1 — COMMENTED** [review]
thanks, look forward to seeing attenuation tested in QB soon, @dnikodem

**Review by btian1 — APPROVED** [review]

**Comment by dnikodem** [comment]
SOFCI TEST

**Comment by kv2019i** [comment]
One known fail in https://sof-ci.01.org/sofpr/PR7793/build9799/devicetest/index.html , otherwise looks good, proceeding with merge.

**Pull request merged**